### PR TITLE
Jv spring boot project hw 5

### PR DESCRIPTION
**Pull request by ihorvasylenko23**

### CODE REVIEW

**Review by AntonZhdanov — APPROVED** [review]

File: src/main/java/mate/academy/repositiry/book/BookSpecificationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    @Autowired
+    private SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParametrers) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParametrers.authors() != null && searchParametrers.authors().length > 0) {
```
Comment (vovasalyha) [inline src/main/java/mate/academy/repositiry/book/BookSpecificationBuilder.java]: You're duplicating a code many times here. For me it's a good candidate for private method

File: src/main/java/mate/academy/repositiry/book/BookSpecificationProviderManager.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(p -> p.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Can`t find correct "
```
Comment (vovasalyha) [inline src/main/java/mate/academy/repositiry/book/BookSpecificationProviderManager.java]: better throw `NoSuchElementException` if required element in collection is not found

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include: 
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: empty line

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Take a look on few comments. Also, before sending to review next time make sure you resolved issue with PRs, because I see here changes from other tasks.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**